### DISCUSSION TITLE
How to configure this nginx proxy to accept self-signed SSL certificates from a container?

### DISCUSSION THREAD

**Discussion by uwla** [evertramos/nginx-proxy-automation#398] [Q&A] [answered]
I work on a project configured to only accept HTTPS connections. For local development, we use self-signed certificates and it works.

I tried to use nginx-proxy-automation to proxy that project, but I got error 500. I had set `VIRTUAL_PROTO=https` and `VIRTUAL_PORT=443` but the issue seems to be nginx proxy not accepting self-signed certificates. So I tried to add `proxy_ssl_verify off' to the `data/conf.d/conf` configuration file and restart the  `proxy-web` container, but as soon as the container restarts it overwrites that file and deletes that line!

What I'm doing when I need to work with this project is to stop nginx-proxy-automation and publish the port 443:443 to the project container.

However, this prevents me from using nginx-proxy-automation which I have been using to work with multiple projects under distinct local domains (i.e, app.test, app2.test, app3.test, and so on).

**Comment by evertramos** [comment]
Hi!

In fact you don´t need any of that... You just add the tls file to the location where is should be, under the domain name, and start the container with TLS activated. It will work it out just fine.

I will send a clear example here later, with CloudFlare TLS.

**Reply by uwla** [reply]
Amazing!

**Comment by uwla** [comment, marked as answer]
Just to update, here is how I solved the problem:

```bash
docker container cp <container_name>:/etc/ssl/private/my.key data/certs/mydomain.test.key
docker container cp <container_name>:/etc/ssl/certs/my.crt data/certs/mydomain.test.crt
```

Where `my.key` and `my.crt` are the certificate files in the container and `mydomain.test` is the local domain.

I just copied the certificates from inside the container (which generated them upon build) to the path where the NGINX proxy stores the certificates.
